api/v1alpha1/edgeconfig_types.go

```diff
@@ -0,0 +1,151 @@
+
+// EdgeConfigSpec defines the desired state of EdgeConfig
+type EdgeConfigSpec struct {
+	// The ansible playbook command to execute
+	EdgePlaybook *EdgePlaybookSpec `json:"edgePlaybook,omitempty"`
+
+	// EdgeDeviceGroupName in which this EdgeConfig is applied
+	EdgeDeviceGroupName string `json:"edgeDeviceGroupName,omitempty"`
+}
+
+// EdgeConfigStatus defines the observed state of EdgeConfig
+type EdgeConfigStatus struct {
+	EdgePlaybookStatus *EdgePlaybookStatus `json:"edgePlaybookStatus,omitempty"`
+}
+
+// EdgePlaybookSpec defines the desired state of EdgePlaybook
+type EdgePlaybookSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	// The ansible playbook command to execute
+	AnsiblePlaybookCmd *AnsiblePlaybookCmd `json:"ansiblePlaybookCmd,omitempty"`
+	//Execution strategy for each playbook
+	ExecutionStrategy map[string]ExecutionStrategy `json:"playbooksPriorityMap,omitempty"`
+}
+
+// EdgePlaybookStatus defines the observed state of EdgePlaybook
+type EdgePlaybookStatus struct {
+	Condition                 EdgePlaybookCondition `json:"condition,omitempty"`
+	LastSeenTime              metav1.Time           `json:"lastSeenTime,omitempty"`
+	LastSyncedResourceVersion string                `json:"lastSyncedResourceVersion,omitempty"`
+}
+
+type AnsiblePlaybookCmd struct {
+	// username who execute the playbook
+	User string `json:"user,omitempty"`
+	// the ansible's playbooks list with execution priority
+	// +kubebuilder:validation:MinProperties=1
+	Playbooks map[string]Playbook `json:"playbooksPriorityMap,omitempty"`
+	// the ansible's playbook options for each playbook
+	Options map[string]*AnsibleOptions `json:"ansibleOptions,omitempty"`
+	// the ansible's playbook privilege escalation options for each playbook
+	PrivilegeEscalationOptions map[string]*PrivilegeEscalationOptions `json:"privilegeEscalationOptions,omitempty"`
+}
+
+type ExecutionStrategy string
+
+const (
+	StopAtFailuire    ExecutionStrategy = "Stop at first failure"
+	ContinueOnFailure ExecutionStrategy = "Continue on failure"
+)
+
+type AnsibleOptions struct {
+	// don't make any changes; instead, try to predict some of the changes that may occur
+	Check bool `json:"check,omitempty"`
+}
+
+type PrivilegeEscalationOptions struct {
+	Become bool `json:"become,omitempty"`
+	// +kubebuilder:validation:Enum=sudo;su
+	BecomeMethod string `json:"becomeMethod,omitempty"`
+	BecomeUser   string `json:"becomeUser,omitempty"`
+}
+
+type Playbook struct {
+	// Playbook content
+	Content string `json:"content"`
+	// The connection timeout on ansible-playbook
+	Timeout uint64 `json:"timeout,omitempty"`
+	// TODO: Enum like linux capabilities ?
+	// The required privelege level necessary to execute the playbook
+	RequiredPrivilegeLevel []string `json:"requiredPrivilegeLevel,omitempty"`
+}
+
+type EdgePlaybookCondition struct {
```

Comment:
I would love to have  documentation for possible condition state sets - not necessarily now, with the controller implementation at the latest.

Reply:
See https://github.com/project-flotta/flotta-operator/pull/145/files#r847465346